src/kayenta/metricStore/signalfx/metricConfigurer.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from 'react';
+import { get } from 'lodash';
+import { Action } from 'redux';
+import { connect } from 'react-redux';
+import FormRow from 'kayenta/layout/formRow';
+import { ICanaryState } from 'kayenta/reducers';
+import * as Creators from 'kayenta/actions/creators';
+import { DisableableInput, DISABLE_EDIT_CONFIG } from 'kayenta/layout/disableable';
+import { ICanaryMetricConfig } from 'kayenta/domain';
+import autoBindMethods from 'class-autobind-decorator';
+import { HelpField } from '@spinnaker/core';
+import KeyValueList, { IKeyValuePair, IUpdateKeyValueListPayload } from '../../layout/keyValueList';
+
+import './metricConfigurer.less'
+import {
+  ICanaryMetricValidationErrors,
+  MetricValidatorFunction
+} from '../../edit/editMetricValidation';
+import { createSelector } from 'reselect';
+import { editingMetricSelector } from '../../selectors';
+
+interface ISignalFxMetricConfigurerStateProps {
+  editingMetric: ICanaryMetricConfig,
+  validationErrors: ICanaryMetricValidationErrors;
+}
+
+interface ISignalFxMetricConfigurerDispatchProps {
+  updateMetricName: (name: string) => void;
+  updateAggregationMethod: (method: string) => void;
+  updateQueryPairs: (payload: IUpdateKeyValueListPayload) => void;
+}
+
+type SignalFxMetricConfigurerProps = ISignalFxMetricConfigurerStateProps & ISignalFxMetricConfigurerDispatchProps;
+
+export const queryFinder = (metric: ICanaryMetricConfig) => get(metric, 'query.metricName', '');
+const getSignalFxMetric = queryFinder;
+const getQueryPairs = (metric: ICanaryMetricConfig) => get(metric, 'query.queryPairs', []) as IKeyValuePair[];
+const getAggregationMethod = (metric: ICanaryMetricConfig) => get(metric, 'query.aggregationMethod', '');
+
+export const SIGNAL_FLOW_SIMPLE = 'signalfx';
+
+@autoBindMethods
+class SignalFxMetricConfigurer extends React.Component<SignalFxMetricConfigurerProps> {
+
+  public onMetricNameChange(e: React.ChangeEvent<HTMLInputElement>) {
+    this.props.updateMetricName(e.target.value);
+  }
+
+  public onAggregationMethodChange(e: React.ChangeEvent<HTMLInputElement>) {
+    this.props.updateAggregationMethod(e.target.value);
+  }
+
+  public render() {
+
+    const { editingMetric, updateQueryPairs, validationErrors } = this.props;
+
+    return (
+      <section>
+        <FormRow
+          label="SignalFx Metric"
+          error={get(validationErrors, 'signalFxMetric.message', null)}
+        >
+          <DisableableInput
+            type="text"
+            value={getSignalFxMetric(editingMetric)}
+            onChange={this.onMetricNameChange}
+            disabledStateKeys={[DISABLE_EDIT_CONFIG]}
+          />
+        </FormRow>
+        <FormRow
```

Comment:
I think `FormRow` has a prop `helpId` that places the `HelpField` next to the form field label.

Reply:
I'm looking here: https://github.com/spinnaker/deck-kayenta/blob/master/src/kayenta/layout/formRow.tsx#L21

(it's just doing what you're already doing)

Reply:
I just pulled, and got those changes, there weren't in my local code base.

- [ ] update